src/app/data-client.test.ts

```diff
@@ -0,0 +1,263 @@
+          cb(null, {
+            getDataList: () => [],
+          });
+        });
+    });
+    test('get binary data', async () => {
+      const promise = await subject().binaryDataByFilter();
+      expect(promise.length).toEqual(2);
+      expect(promise[0]?.binary).toEqual(bin1);
+      expect(promise[1]?.binary).toEqual(bin2);
+    });
+
+    test('get filtered binary data', async () => {
+      const filter = subject().createFilter({
+        componentName: 'testComponentName',
+        componentType: 'testComponentType',
+      });
+
+      const dataReq = new DataRequest();
+      dataReq.setFilter(filter);
+      dataReq.setLimit(100);
+      dataReq.setLast('');
+      const req = new BinaryDataByFilterRequest();
+      req.setDataRequest(dataReq);
+      req.setCountOnly(false);
+
+      await subject().binaryDataByFilter(filter);
+      expect(methodSpy).toHaveBeenCalledWith(
+        req,
+        expect.anything(),
+        expect.anything()
+      );
+    });
+  });
+
+  describe('binaryDataById tests', () => {
+    let methodSpy: SpyInstance;
+    beforeEach(() => {
+      methodSpy = vi
+        .spyOn(DataServiceClient.prototype, 'binaryDataByIDs')
+        // @ts-expect-error compiler is matching incorrect function signature
+        .mockImplementation((_req: BinaryDataByIDsRequest, _md, cb) => {
+          cb(null, binDataResponse);
+        });
+    });
+
+    const binaryId1 = new BinaryID();
+    binaryId1.setFileId('testFileId1');
+    binaryId1.setOrganizationId('testOrgId');
+    binaryId1.setLocationId('testLocationId');
+    const binaryId2 = new BinaryID();
+    binaryId2.setFileId('testFileId1');
+    binaryId2.setOrganizationId('testOrgId');
+    binaryId2.setLocationId('testLocationId');
+
+    test('get binary data by ids', async () => {
+      const promise = await subject().binaryDataByIds([
+        binaryId1.toObject(),
+        binaryId2.toObject(),
+      ]);
+      expect(promise.length).toEqual(2);
+      expect(promise[0]?.binary).toEqual(bin1);
+      expect(promise[1]?.binary).toEqual(bin2);
+    });
+
+    test('get binary data by id', async () => {
+      const req = new BinaryDataByIDsRequest();
+      req.setBinaryIdsList([binaryId1]);
+      req.setIncludeBinary(true);
+
+      await subject().binaryDataByIds([binaryId1.toObject()]);
+      expect(methodSpy).toHaveBeenCalledWith(
+        req,
+        expect.anything(),
+        expect.anything()
+      );
+    });
+  });
+
+  describe('createFilter tests', () => {
+    test('create empty filter', () => {
+      const testFilter = subject().createFilter({});
+      expect(testFilter).toEqual(new Filter());
+    });
+
+    test('create filter', () => {
+      const opts = { componentName: 'camera' };
+      const testFilter = subject().createFilter(opts);
+
+      const actualFilter = new Filter();
+      actualFilter.setComponentName('camera');
+
+      expect(testFilter).toEqual(actualFilter);
```

Comment:
[minor] we should rename "actualFilter" to something like "expectedFilter" - I'm used to "actual" referring to the result of the subject function that we defined, not the the value we say it should be equal to.

Same goes for all of the tests below if you agree.

```suggestion
      const testFilter = subject().createFilter(opts);

      const expectedFilter = new Filter();
      expectedFilter.setComponentName('camera');

      expect(testFilter).toEqual(expectedFilter);
```